scipy/sparse/compressed.py

```diff
@@ -59,22 +60,25 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False):
                     maxval = None
                     if shape is not None:
                         maxval = max(shape)
-                    idx_dtype = get_index_dtype((indices, indptr), maxval=maxval, check_contents=True)
+                    idx_dtype = get_index_dtype((indices, indptr),
+                                                maxval=maxval,
+                                                check_contents=True)
 
-                    self.indices = np.array(indices, copy=copy, dtype=idx_dtype)
+                    self.indices = np.array(indices, copy=copy,
+                                            dtype=idx_dtype)
                     self.indptr = np.array(indptr, copy=copy, dtype=idx_dtype)
                     self.data = np.array(data, copy=copy, dtype=dtype)
                 else:
-                    raise ValueError("unrecognized %s_matrix constructor usage" %
-                            self.format)
+                    raise ValueError("unrecognized {}_matrix constructor usage"
+                                     "".format(self.format))
```

Comment:
This line-wrapping is pretty awkward. I think I prefer the %-formatted way (here and elsewhere).

Reply:
That became a habit out of a few PRs that had dangling arguments behind the actual arg e.g., 

    raise ValueError('some text %s and some more %s` % text1, text2)

would lead to errors and often exceptions are not tested.

 In my opinion noticing where the argument would show up via `{}` is a bit more readable on instead of parsing `%s_matrix`.  But I don't have any strong opinion on it.

Reply:
I just don't like the hanging `"".format(...)` on its own line, but I get the rationale so feel free to ignore me here.